File: p12_Mirror_Incremental_Number_Triangle/main.ts
const input = require("prompt-sync")();

let n: number = Number(input())

for(let i = 1; i <= n; i++){
    let rowContent: string = ""

    for(let j = 0; j < i; j++){
        rowContent += i + j + " ";
    }
    rowContent = rowContent.trim();

    let lastRow: String = ""
    for(let j = 0; j < n; j++){
        lastRow += n + j + " "
    }

    lastRow = lastRow.trim()

    let spaces: string = " ".repeat(lastRow.length - rowContent.length);
    console.log(spaces + rowContent);
}

File: p12_Mirror_Incremental_Number_Triangle/Main.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int n = input.nextInt();

        for (int i = 1; i <= n; i++) {
            String rowContent = "";

            for (int j = 0; j < i; j++) {
                rowContent += (i + j) + " ";
            }

            rowContent = rowContent.trim();

            String lastRow = "";

            for (int j = 0; j < n; j++) {
                lastRow += (n + j) + " ";
            }

            lastRow = lastRow.trim();

            String spaces = " ".repeat(lastRow.length() - rowContent.length());

            System.out.println(spaces + rowContent);
        }

        input.close();
    }
}